### ISSUE TITLE
MVP Sierra integration

### ISSUE THREAD

**Issue by mikeapp** [yalelibrary/YUL-DC#2498]
**Story**

Sierra should be added as a metadata source in DCS.   This work should be done in a feature branch.

Prerequisite: #2499 

**Acceptance**
- [x] Add Sierra as an Authoritative Metadata Source in Management DB
  - [x] The Metadata Cloud name for this source should be `sierra` and the display name `Sierra` 
- [x] Batch jobs that accept a metadata source should accept this new code 
- [x] Metadata should be fetched from the MC endpoint for Sierra
   - [x] MC API URL is dependent on #2499 
- [x] Review codebase for other string references to metadata sources that might need to be changed.
   - [x] Looks like a literal array is used in the data tables (refactor?)

**Comment by jpengst** [comment]
PR: https://github.com/yalelibrary/yul-dc-management/pull/1207

**Comment by mikeapp** [comment]
Comments on PR, also created #2514 for future work

**Comment by mikeapp** [comment]
"Review on Demo"

**Comment by MaggieZhaoYale** [comment]
Example of Sierra metadata cloud: 
https://metadata-api-uat.library.yale.edu/metadatacloud/api/sierra/bib/273496?mediaType=json
or https://metadata-api-uat.library.yale.edu/metadatacloud/api/# version/sierra/bib/273496?mediaType=json

**Comment by sshetenhelm** [comment]
Testing on demo using '[create parent objects' ](https://collections-demo.library.yale.edu/management/batch_processes/111/parent_objects/30000061)batch process.

- [Parent object created](https://collections-demo.library.yale.edu/management/parent_objects/30000061)
- Metadata Cloud provides json via link
- No json in object

Received following error message:
~~~
Setup job failed to save: undefined method `siera_json=' for #<ParentObject:0x0000564241660df8> Did you mean? sierra_json= sierra_json sierra_json?
~~~

**Comment by jpengst** [comment]
Oh, whoops! Typo in `siera_json` :(

**Comment by jpengst** [comment]
Typo PR: https://github.com/yalelibrary/yul-dc-management/pull/1210

**Comment by jpengst** [comment]
Trying to test this on demo but Solr seems to be down. Received a 500 Solr error when trying to delete an object. Create Parent Batch Process also reports 500 Solr.

**Comment by jpengst** [comment]
Testing on demo. Batch fails for `undefined method zero?`  


![Image](https://github.com/yalelibrary/YUL-DC/assets/24666568/e7be1db5-7c98-4198-9aca-e7b000acc263)

  
I think this is being thrown in the `sierra_json` method for `itemID`

**Comment by jpengst** [comment]
PR for itemID check: https://github.com/yalelibrary/yul-dc-management/pull/1217

**Comment by jpengst** [comment]
bug fix deployed to Demo

**Comment by sshetenhelm** [comment]
[Sierra JSON present for parent created through batch process (no children so no SOLR)](https://collections-demo.library.yale.edu/management/parent_objects/30000061)
Sierra JSON present for existing parent object attached to Sierra. [MOSTLY Sierra metadata displays in Blacklight](https://collections-demo.library.yale.edu/catalog/2000208), HOWEVER under the 'identifiers' section, the Law bib of 273496 make a /10079/bibid handle that points to the Orbis bib of the same number (SEE: [273496](http://hdl.handle.net/10079/bibid/273496) )

**Comment by jpengst** [comment]
I'm looking at this this morning and noticed this Sierra items Identifier section with the Orbis Record id and link is consistent with other objects Identifier section on the site. Should the Orbis Record number and link be something different?

**Comment by sshetenhelm** [comment]
If an object has Sierra as its data source: 

- Front end should display "Morris Record" instead of Orbis record
- Created link should have prefix of https://morris.law.yale.edu/record=b (so full link would be ex. [https://morris.law.yale.edu/record=b](https://morris.law.yale.edu/record=b273496))

**Comment by jpengst** [comment]
![Image](https://github.com/yalelibrary/YUL-DC/assets/24666568/03e8c22e-36bb-4856-aab8-2e00e19d6cfa)

**Comment by jpengst** [comment]
Deployed to Demo
